### PR TITLE
jobsdb cache fix

### PR DESCRIPTION
**Pull request by chandumlg**
**Fixes** # 
Cache issues in multitenant scenario

## Type of change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added unit tests for the code
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: jobsdb/jobsdb_export.go [hunk 1]
```diff
@@ -112,9 +113,13 @@ func (jd *HandleT) GetNonMigratedAndMarkMigrating(count int) []*JobT {
 	jd.assertError(err)
 
 	for ds, updatedStatesByCustomer := range updatedStatesByDS {
+		allUpdatedStates := make([]string, 0)
```
Comment (lvrach) [inline jobsdb/jobsdb_export.go:116]: Maybe we should consider using https://github.com/deckarep/golang-set in such cases. I was also considering using this library for dedup as well. I am not sure of the performance benefits though.

File: jobsdb/jobsdb.go [hunk 2]
```diff
@@ -1965,20 +1975,14 @@ type cacheEntry struct {
...
-
 	jd.dsCacheLock.Lock()
 	defer jd.dsCacheLock.Unlock()
 
 	//This means we want to mark/clear all customVals and stateFilters
 	//When clearing, we remove the entire dataset entry. Not a big issue
 	//We process ALL only during internal migration and caching empty
 	//results is not important
-	if len(stateFilters) == 0 || len(customValFilters) == 0 || customer == `` {
+	if len(stateFilters) == 0 || len(customValFilters) == 0 {
```
Comment (lvrach) [inline jobsdb/jobsdb.go:1985]: Why we no longer check for `customer == '' `

Comment (chandumlg) [reply jobsdb/jobsdb.go:1985]: Ideally we should never receive customer as empty string.

**Review by cisse21 — APPROVED** [review]

**Comment by cisse21** [comment]
Do we still need this interface ? `CacheOperator` or can we get rid of it?

**Review by lvrach — APPROVED** [review]

**Review by Sidddddarth — APPROVED** [review]

**Pull request merged**